### ISSUE TITLE
x, y or rho, theta for AffectNet Emotions

### ISSUE THREAD

**Issue by GiannisPikoulis** [stefanodapolito/GANmut#5]
Hello. 

I want to test your method for image emotion editing. How do I find the correspondance between discrete emotions and the model variables (x, y, rho, theta) in order to use them with the `images_generation.ipynb` notebook?

**Comment by stefanodapolito** [comment]
Hi. Yes, I should have put an image of the learned map. 
To get the mean (i.e. the x, and y you are searching) and covariance matrices for each emotion for gaussian model :
`G = GANmut(G_path='../learned_generators/gaus_2d/1800000-G.ckpt',model='gaussian') `
`G.G.print_expr()`
To get the unit vectors representing each emotion for the linear model
`G = GANmut(G_path='../learned_generators/lin_2d/1000000-G.ckpt',model='linear')`
`G.G.print_axes()`
you will obtain the unit vector representing each emotion. 
The order is # 0: Neutral, 1: Happy, 2: Sad, 3: Surprise, 4: Fear, 5: Disgust, 6: Anger

**Comment by GiannisPikoulis** [comment]
So, for example in the Linear Model, do I have to take the first row of `G.G.print_axes()` --> `(theta_happy, rho_happy)` and pass it as input arguments along with the source image to `G.emotion_edit`. I think this is wrong cause `G.emotion_edit` takes as input `theta, rho` and computes `[cos(theta)*rho, sin(theta)*rho]`, while G.G.print_axes() returns the Cartesian coordinates by default. I think you need to rethink the implementation of your notebook demo.

**Comment by stefanodapolito** [comment]
Exactly, to get `theta_emotion` you would need to do atan2(x,y) where x,y are coordinate of the unit vectors you obtain from print_axes(). `rho_emotion` is your choice,  it depends on the intensity you want.
Thanks to point out. When I get a bit of spare time I will update the notebook with all the necessary coordinates

**Comment by GiannisPikoulis** [comment]
I think it should be atan2(y, x) assuming you use something like `numpy.arctan2`. Please correct me if I'm wrong. I've assumed that `x: v[0], y: v[1]` where v is a row of `G.G.print_axes()`.

And one last thing in order to be sure. When using the Gaussian model, I just pass the corresponding row coordinates of `tanh(G.G.mu)` to the `x, y` arguments of `G.emotion_edit`.

Thanks in advance.
